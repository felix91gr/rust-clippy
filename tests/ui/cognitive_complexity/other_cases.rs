#![allow(clippy::all)]
#![warn(clippy::cognitive_complexity)]
#![allow(unused)]

fn main() {}

/*
    Other Tests
*/

#[clippy::cognitive_complexity = "1"]
fn nested_functions_are_counted(n: usize) -> usize {
    fn lucas_number(n: usize) -> usize {
        match n {
            0 => 2,
            1 => 1,
            _ => lucas_number(n - 1) + lucas_number(n - 2),
        }
    }
    lucas_number(n)
}
