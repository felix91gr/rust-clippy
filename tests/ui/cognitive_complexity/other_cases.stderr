error: the function has a cognitive complexity of 40
  --> $DIR/other_cases.rs:12:1
   |
LL | / fn nested_functions_are_counted(n: usize) -> usize {
LL | |     fn lucas_number(n: usize) -> usize {
LL | |         match n {
LL | |             0 => 2,
...  |
LL | |     lucas_number(n)
LL | | }
   | |_^
   |
   = note: `-D clippy::cognitive-complexity` implied by `-D warnings`
   = help: you could split it up into multiple smaller functions

error: aborting due to previous error

